lib/_internal/file_picker_web.dart

```diff
@@ -208,7 +208,33 @@ class FilePickerWeb extends FilePicker {
       final blob = file.slice(start, end);
       reader.readAsArrayBuffer(blob);
       await EventStreamProviders.loadEvent.forTarget(reader).first;
-      yield reader.result as List<int>;
+      final JSAny? readerResult = reader.result;
+      if (readerResult == null) {
+        continue;
+      }
+      if (readerResult is ByteBuffer) {
+        // An error may occur:
+        // TypeError: Instance of 'NativeByteBuffer': type 'NativeByteBuffer' is not a subtype of type 'List<int>'.
+        // Therefore, we need to make a type discrimination of the reader.result.
+        yield (readerResult as ByteBuffer).asUint8List();
+        start += _readStreamChunkSize;
+        continue;
+      }
+      // Handle byte buffers.
+      if (readerResult.instanceOfString('JSArrayBuffer')) {
+        yield (readerResult as JSArrayBuffer).toDart.asUint8List();
+        start += _readStreamChunkSize;
+        continue;
+      }
+      // Handle JS Array type.
+      if (readerResult.instanceOfString('Array')) {
+        // Assume this is a List<int>.
+        yield (readerResult as JSArray).toDart.cast<int>();
+        start += _readStreamChunkSize;
+        continue;
+      }
+      // Default this is a List<int>
```

Comment:
You already handled the `List<int>` case with `if (readerResult.instanceOfString('Array')) {` on line 230, so this block can be removed.